flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskExecutorLocalStateStoresManager.java

```diff
@@ -89,6 +90,22 @@ public TaskExecutorLocalStateStoresManager(
 				}
 			}
 		}
+
+		// install a shutdown hook
+		this.shutdownHook = new Thread("TaskExecutorLocalStateStoresManager shutdown hook") {
```

Comment:
It might make sense to reuse `BlobUtils#addShutdownHook` or to refactor this and then reuse.

Reply:
I made a separate commit that introduces `ShudownHookUtil`. Used it to deduplicated the same code in like 20 places...